### ISSUE TITLE
Cannot build documentation and skip notebooks

### ISSUE THREAD

**Issue by astrojuanlu** [poliastro/poliastro#1186]
If I try `POLIASTRO_SKIP_NOTEBOOKS=True tox -e docs -v`, I get errors like

```
Traceback (most recent call last):                                                                                     
  File "/home/juanlu/Projects/LSF/poliastro/library/.tox/docs/lib/python3.8/site-packages/sphinx/cmd/build.py", line 280, in build_main
    app.build(args.force_all, filenames)                                                                               
  File "/home/juanlu/Projects/LSF/poliastro/library/.tox/docs/lib/python3.8/site-packages/sphinx/application.py", line 350, in build
    self.builder.build_update()
  File "/home/juanlu/Projects/LSF/poliastro/library/.tox/docs/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 292, in build_update
    self.build(to_build,            
  File "/home/juanlu/Projects/LSF/poliastro/library/.tox/docs/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 306, in build
    updated_docnames = set(self.read())
  File "/usr/lib/python3.8/contextlib.py", line 120, in __exit__
    next(self.gen)                                                                                                                                                                                                                            
  File "/home/juanlu/Projects/LSF/poliastro/library/.tox/docs/lib/python3.8/site-packages/sphinx/util/logging.py", line 213, in pending_warnings
    memhandler.flushTo(logger)                                                                                                                                                                                                                
  File "/home/juanlu/Projects/LSF/poliastro/library/.tox/docs/lib/python3.8/site-packages/sphinx/util/logging.py", line 178, in flushTo
    logger.handle(record)                                                                                                                                                                                                                     
  File "/usr/lib/python3.8/logging/__init__.py", line 1587, in handle
    self.callHandlers(record)                                                                                                                                                                                                                 
  File "/usr/lib/python3.8/logging/__init__.py", line 1649, in callHandlers
    hdlr.handle(record)                                                                                                
  File "/usr/lib/python3.8/logging/__init__.py", line 946, in handle
    rv = self.filter(record)                                                                                                                                                                                                                  
  File "/usr/lib/python3.8/logging/__init__.py", line 807, in filter
    result = f.filter(record)                                                                                                                                                                                                                 
  File "/home/juanlu/Projects/LSF/poliastro/library/.tox/docs/lib/python3.8/site-packages/sphinx/util/logging.py", line 423, in filter
    raise exc                                                                                                          
sphinx.errors.SphinxWarning: /home/juanlu/Projects/LSF/poliastro/library/docs/source/examples/Analyzing NEOs.mystnb:"nbsphinx-thumbnail": No outputs in cell 6
                                                                                                                       
Warning, treated as error:
/home/juanlu/Projects/LSF/poliastro/library/docs/source/examples/Analyzing NEOs.mystnb:"nbsphinx-thumbnail": No outputs in cell 6
```

So we should either:

- Find out if we are misusing nbsphinx,
- Or whether this is a bug upstream,
- Or wether we should remove the option to skip the notebooks.

**Comment by DhruvSondhi** [comment]
Hello @astrojuanlu, I think I may have a lead for the same ... What actually is happening is that because we are using `:tags: [nbsphinx-thumbnail]` for generating the the `nbsphinx` thumbnail for the gallery, if we don't consider the notebooks this cell doesn't get executed & hence no thumbnail is being generated 😥  ... This will also render the gallery without any thumbnails for the notebooks 🤔

**Comment by astrojuanlu** [comment]
Thanks for investigating this @DhruvSondhi ! Maybe we should see if we can temporarily disable the thumbnail generation when notebooks are disabled, or fill them with a dummy thumbnail, allow warnings if not running the notebooks (probably the easiest one), or just remove the possibility of not running the notebooks.

**Comment by DhruvSondhi** [comment]
@astrojuanlu I ran the build without the _warnings turning into errors_ for docs & Yes, I was correct 
![image](https://user-images.githubusercontent.com/66117751/116775875-1d10a580-aa83-11eb-9a0b-69211b58a75f.png)
Now, what I was thinking is that I can generate the `.png` graphs for these notebooks & hence link them to the gallery via the `nbsphinx thumbnail` (as already been done in some notebooks) ... in that way, we do remove the functionality that if any change to the code in the notebook is done, it will not generate the updated thumbnails but it will fix this issue :)
Another comparison can be seen here:
![image](https://user-images.githubusercontent.com/66117751/116775928-7547a780-aa83-11eb-97c0-11b843cf0480.png)

**Comment by mgeier** [comment]
You might be able to silence the warning (and therefore hopefully not triggering the mentioned error) by setting this:

```python
suppress_warnings = ['nbsphinx.thumbnail']
```

**Comment by mgeier** [comment]
PS: I guess this can also be specified on the command line if that makes more sense, but it might need a few iterations to find the right command line syntax. Probably something like `-D suppress_warnings=nbsphinx.thumbnail` could work? Multiple strings can be separated by commas (but without spaces).

**Comment by astrojuanlu** [comment]
Your first option worked, and your second option also worked 😍 I have Sphinx options on our `conf.py` that I had forgotten about, and also I didn't know about `-D`. Thanks a lot @mgeier! What about something like this?

```diff
diff --git a/docs/source/conf.py b/docs/source/conf.py
index f4835fec..e1d5088e 100644
--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -158,6 +158,7 @@ intersphinx_mapping = {
 
 if os.environ.get('POLIASTRO_SKIP_NOTEBOOKS') == 'True':
     nbsphinx_execute = 'never'
+    suppress_warnings.append('nbsphinx.thumbnail')
 else:
     nbsphinx_execute = 'always' 

```

**Comment by mgeier** [comment]
This looks reasonable. Does it work as expected?

**Comment by astrojuanlu** [comment]
It works locally for me, yes! Leaving this as a "good first issue" so any newcomer can open a pull request with the change.

**Comment by sreyan-ghosh** [comment]
Hi @astrojuanlu I am a "newcomer", I would like to work on adding this. I enjoyed reading through your conversation and also got to know about sphinx and astropy. Could you please assign this to me?

Also, just to be clear, I will be making the change [here](https://github.com/poliastro/poliastro/blob/main/docs/source/conf.py) right?

**Comment by astrojuanlu** [comment]
Closed in #1325 , thanks @sreyan-ghosh !
